src/cache/cache-manager.ts

```diff
@@ -69,49 +70,135 @@ export class CacheManager {
 
   async get(
     cacheKey: CacheKey,
-    expiryAdjustmentSeconds = DEFAULT_EXPIRY_ADJUSTMENT_SECONDS
+    options: {
+      expiryAdjustmentSeconds: number,
+    } = {
+        expiryAdjustmentSeconds: DEFAULT_EXPIRY_ADJUSTMENT_SECONDS,
+      },
   ): Promise<Partial<CacheEntry> | undefined> {
-    let wrappedEntry = await this.cache.get<WrappedCacheEntry>(
-      cacheKey.toKey()
+    const activeToken = await this.getActiveToken(
+      cacheKey,
+      options.expiryAdjustmentSeconds,
     );
 
-    if (!wrappedEntry) {
-      const keys = await this.getCacheKeys();
+    if (activeToken) {
+      return activeToken.body;
+    }
 
-      if (!keys) return;
+    const inactiveToken = await this.getInactiveToken(
+      cacheKey,
+      options.expiryAdjustmentSeconds,
+    );
 
-      const matchedKey = this.matchExistingCacheKey(cacheKey, keys);
+    if (inactiveToken) {
+      return inactiveToken.body;
+    }
 
-      if (matchedKey) {
-        wrappedEntry = await this.cache.get<WrappedCacheEntry>(matchedKey);
-      }
+    const keys = await this.getCacheKeys();
+
+    if (!keys) return;
+
+    const siblingToken = await this.getSiblingToken(
+      cacheKey,
+      keys,
+      options.expiryAdjustmentSeconds,
+    );
+
+    if (siblingToken) {
+      return siblingToken.body;
     }
 
-    // If we still don't have an entry, exit.
-    if (!wrappedEntry) {
-      return;
+    return;
+  }
+
+  async onNoRefreshableToken(cacheKey: CacheKey): Promise<undefined> {
+    await this.cache.remove(cacheKey.toKey());
+    await this.keyManifest?.remove(cacheKey.toKey());
+    return;
+  }
+
+  async getRefreshToken(entry: WrappedCacheEntry, cacheKey: CacheKey): Promise<WrappedCacheEntry | undefined> {
+    if (!entry.body.refresh_token) {
+      return this.onNoRefreshableToken(cacheKey);
     }
 
-    const now = await this.nowProvider();
-    const nowSeconds = Math.floor(now / 1000);
+    entry.body = {
+      refresh_token: entry.body.refresh_token,
+    };
 
-    if (wrappedEntry.expiresAt - expiryAdjustmentSeconds < nowSeconds) {
-      if (wrappedEntry.body.refresh_token) {
-        wrappedEntry.body = {
-          refresh_token: wrappedEntry.body.refresh_token
-        };
+    await this.cache.set(cacheKey.toKey(), entry);
```

Comment:
So we have a getRefreshToken method that updates the cache and removed everything but that refreshToken from the body? Is that expected when calling something that is called `getRefreshToken`?